### PR TITLE
Module Import Test

### PR DESCRIPTION
**Pull request by dkress59**
I am totally open to removing the Dockerfile. The idea was just to fail early, e.g. by working this local test into the prePublish script. I added it in the first place to set up the basic structure of the cra.yml workflow – and I do see that having docker in here as an external dependency is probably not the way to go.

### CODE REVIEW

**Comment by dkress59** [comment]
includes `index.d.ts` from https://github.com/react-async-storage/react-async-storage/pull/20

**Comment by Goldziher** [comment]
Ok, so long story short - I think we can do a similar test in the regular tests and do not to use the entire CRA stuff to do this. In fact, I already test the StorageProvider quite thoroughly - as well as the useStorage hook etc. Please take a look at the tests.

Now for the index.d.ts - this is not required. The only time where it is required is when you have a legacy JS module and want to declare it in ts, but an npm module that is properly types, as this library is, does not require this. 

Its true that in the kind of manipulation you are making using the CRA script you will need to declare the module for TS to work normally, but it's because you are not importing it as an ES (ESM) or CJS module. 

In other words - we should close this PR.
